staging/src/k8s.io/component-base/version/version.go

```diff
@@ -134,15 +136,23 @@ func (m *effectiveVersion) Validate() []error {
 	emulationVersion := m.emulationVersion.Load()
 	minCompatibilityVersion := m.minCompatibilityVersion.Load()
 
-	// emulationVersion can only be 1.{binaryMinor-1}...1.{binaryMinor}.
+	// emulationVersion can only be 1.{binaryMinor-3}...1.{binaryMinor}
+	// with a floor of 1.31.
 	maxEmuVer := binaryVersion
-	minEmuVer := binaryVersion.SubtractMinor(1)
+	minEmuVer := binaryVersion.SubtractMinor(3)
+	if minEmuVer.LessThan(floorVersion) {
+		minEmuVer = floorVersion
+	}
 	if emulationVersion.GreaterThan(maxEmuVer) || emulationVersion.LessThan(minEmuVer) {
 		errs = append(errs, fmt.Errorf("emulation version %s is not between [%s, %s]", emulationVersion.String(), minEmuVer.String(), maxEmuVer.String()))
 	}
-	// minCompatibilityVersion can only be 1.{binaryMinor-1} for alpha.
+	// minCompatibilityVersion can only be 1.{binaryMinor-3} to 1.{binaryMinor-1}
```

Comment:
nit: I'm a bit confused in these tests why minCompatibilityVersion is a parameter?  Shouldn't this just be calculated from binaryMinor as mentioned in the comment here (vs. set by anything directly)?